content/en/docs/guides/compression.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Compression
+description: >-
+    How to use compression to reduce the bandwidth used by gRPC
+---
+
+### Overview
+
+Compression is used to reduce the amount of bandwidth used between peers. The compression supported by gRPC acts at the individual message level.
+
+Depends on language, it's possible to control compression settings at channel, call and message level. Different language also support different compression algorithms including customized compressor. 
+
+### Compression Method Asymmetry Between Peers
+
+gRPC allows asymmetric compression communication, whereby a response may be compressed differently, if at all. A gRPC peer may choose to respond using a different compression method to that of the request, including not performing any compression, regardless of channel and RPC settings (for example, if compression would result in small or negative gains).
+
+If a client message is compressed by an algorithm that is not supported by a server, the message will result in an `UNIMPLEMENTED` error status on the server. The server will then include a `grpc-accept-encoding` response header which specifies the algorithms that the server accepts.
+
+If the client message is compressed using one of the algorithms from the `grpc-accept-encoding` header and an `UNIMPLEMENTED` error status is returned from the server, the cause of the error won't be related to compression. 
+
+If a server sent data which is compressed by an algorithm that is not supported by the client, an `INTERNAL` error status will occur on the client side.
+
+Note that a peer may choose to not disclose all the encodings it supports. However, if it receives a message compressed in an undisclosed but supported encoding, it will include said encoding in the response's `grpc-accept-encoding` header.
+
+For every message a server is requested to compress using an algorithm it knows the client doesn't support (as indicated by the last `grpc-accept-encoding` header received from the client), it will send the message uncompressed.
+
+### Specific Disabling of Compression
+
+If the user requests to disable compression, the next message will be sent uncompressed. This is instrumental in preventing BEAST/CRIME attacks. This applies to both the unary and streaming cases.
```

Comment:
wikipedia link for BEAST/CRIME attacks

Reply:
Is the BEAST attack applicable here? I though it only applies to SSL/TLS